src/TunNetCom.SilkRoadErp.Sales.Api/Features/Products/CreateProduct/CreateProductCommandHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace TunNetCom.SilkRoadErp.Sales.Api.Features.Products.CreateProduct;
+
+public class CreateProductCommandHandler(SalesContext _context, ILogger<CreateProductCommandHandler> _logger) : IRequestHandler<CreateProductCommand, Result<string>>
+{
+    public async Task<Result<string>> Handle(CreateProductCommand createProductCommand, CancellationToken cancellationToken)
+    {
+        _logger.LogEntityCreated("Product", createProductCommand);
+
+        var isProductNameExist = await _context.Produit.AnyAsync(
+            pro => pro.Nom == createProductCommand.Nom,
+            cancellationToken);
+
+        if (isProductNameExist)
+        {
+            return Result.Fail("product_name_exist");
+        }
```

Comment:
Ask Nieze about this method isProductNameExist, because you usually work with Product
This method is typically applied to Refe

Reply:
@Mayssabenjoud @khiari1 yes we must look for both : Nom and Refe